Refresher/Patching/Patcher.cs

```diff
@@ -81,24 +167,26 @@ public IEnumerable<Message> Verify(string url)
 
     public void PatchUrl(string url)
     {
-        using MemoryStream ms = new(this.Data);
-        using BinaryWriter writer = new(ms);
+        using BinaryWriter writer = new(this.Stream);
 
-        // Using BinaryWriter.Write writes a length-oriented string, get this for null terminated instead
-        byte[] urlAsBytes = Encoding.Default.GetBytes(url);
+        // Get a null-terminated byte sequence of the URL, as BinaryWriter.Write(string) writes a length-prepended string,
+        byte[] urlAsBytes = Encoding.UTF8.GetBytes(url);
 
-        if (this._httpUrlInfo.Value.HasValue) PatchUrlFromInfo(writer, this._httpUrlInfo.Value.Value, urlAsBytes);
-        if (this._httpsUrlInfo.Value.HasValue) PatchUrlFromInfo(writer, this._httpsUrlInfo.Value.Value, urlAsBytes);
+        if (this._targets.Value.Count != 0) PatchUrlFromInfoList(writer, this._targets.Value, urlAsBytes);
     }
 
-    private static void PatchUrlFromInfo(BinaryWriter writer, PatchTargetInfo info, byte[] url)
+    private static void PatchUrlFromInfoList(BinaryWriter writer, List<PatchTargetInfo> targets, byte[] url)
     {
-        writer.BaseStream.Position = info.Offset;
-        writer.Write(url);
-        
-        for (int i = 0; i < info.Length - url.Length; i++)
+        foreach (PatchTargetInfo target in targets)
         {
-            writer.Write('\0');
+            if (url.Length > target.Length)
+                throw new ArgumentException("URL is too long to fit in the EBOOT!", nameof(url));
```

Comment:
This *can* stay, but it should definitely be checked first in `Verify()` to properly report the problem to the user before patching.

Reply:
We actually do that in Verify, don't we? `_targets` is a lazy variable, so the actual URL-finding magic happens right there, as we check the count of it.

Reply:
ah well, you are right, didnt realize that, ill add a check for that